server/test/lib/device/device.getDeviceFeaturesStates.test.js

```diff
@@ -0,0 +1,91 @@
+const EventEmitter = require('events');
+const { expect, assert } = require('chai');
+const uuid = require('uuid');
+const { fake } = require('sinon');
+const db = require('../../../models');
+const Device = require('../../../lib/device');
+const Job = require('../../../lib/job');
+
+const event = new EventEmitter();
+const job = new Job(event);
+
+const now = new Date('2000-06-15T03:59:00.000Z');
+const insertStates = async () => {
+  const queryInterface = db.sequelize.getQueryInterface();
+  const deviceFeatureStateToInsert = [];
+  const statesToInsert = 3 * 60;
+  for (let i = 0; i < statesToInsert; i += 1) {
+    const startAt = new Date(now.getTime() - 3 * 60 * 60 * 1000);
+    const date = new Date(startAt.getTime() + ((3 * 60 * 60 * 1000) / statesToInsert) * i);
+    deviceFeatureStateToInsert.push({
+      id: uuid.v4(),
+      device_feature_id: 'ca91dfdf-55b2-4cf8-a58b-99c0fbf6f5e4',
+      value: i,
+      created_at: date,
+      updated_at: date,
+    });
+  }
+  await queryInterface.bulkInsert('t_device_feature_state', deviceFeatureStateToInsert);
+};
+
+describe('Device.getDeviceFeaturesStates', function Describe() {
+  this.timeout(5000);
+
+  afterEach(async () => {
+    const queryInterface = db.sequelize.getQueryInterface();
+    await queryInterface.bulkDelete('t_device_feature_state');
+  });
+  it('should return states between 01:10 and 02:09 with a target between 2000-06-15 00:10 and now using take and skip , only created_at and values', async () => {
```

Comment:
The test assertion seems to test a very specific use case, and I don't see anything in the test code that checks that it's working as advertised?